### ISSUE TITLE
Real-Time localization

### ISSUE THREAD

**Issue by OmarLaz97** [xdspacelab/openvslam#326]
Thank you for your work !!
Is there any methods available to run image localization with real time frames ?

**Comment by ymd-stella** [comment]
You can use the ROS wrapper of openvslam (run_slam, run_localization) to run with real time frames.
See https://openvslam.readthedocs.io/en/master/ros_package.html

**Comment by Zumbalamambo** [comment]
@OmarLaz97 what do you mean by realtime frames?

**Comment by rebellojason7** [comment]
I think he means at frame rate ? I tried running the ros versions in localization mode and 20Hz images, but it doesnt seem to keep up with frame rate ? Any thoughts ?

**Comment by YujiElfahkrany** [comment]
@rebellojason7 did you try reducing the image size?

**Comment by rebellojason7** [comment]
@YujiElfahkrany Thanks for the suggestion, but I am using the same image size for slam and localization, so I dont think the image size makes a difference. Also my image size is 752x480. What I did notice is that there is far more cpu usage when running localization only as opposed to slam. I havent gone through the code but cant imaging why localization would use so much more than slam 

SLAM usage 
![slam](https://user-images.githubusercontent.com/15329203/85632919-36b4b580-b646-11ea-994d-88df6fc96cc7.png)

Localization usage
![localization](https://user-images.githubusercontent.com/15329203/85632923-39afa600-b646-11ea-8117-f968705d5c60.png)

**Comment by YujiElfahkrany** [comment]
@rebellojason7 yes I also noticed this issue, it converges tho, after sometime. 
I think its stuck in some kind of pose optimization

please check this issue for reference  
https://github.com/xdspacelab/openvslam/issues/338

**Comment by rebellojason7** [comment]
@YujiElfahkrany Thanks for pointing me to the thread. Any resolution on it ?

**Comment by rebellojason7** [comment]
@YujiElfahkrany any luck with this issue ?

**Comment by YujiElfahkrany** [comment]
@rebellojason7 for me I launched pose.optimize() from [here](https://github.com/xdspacelab/openvslam/blob/master/src/openvslam/optimize/pose_optimizer.cc#L97) in a different thread and interrupted that thread if it took too much time. 
for me that was a temporary fix

**Comment by YujiElfahkrany** [comment]
I think Sumikura san is testing it here #338
